BUG: Capability Resources error when no Notes are supplied

**When we try to apply :**

resource "sonatyperepo_capability_ui_branding" "main" {
  enabled = true

  properties = {
    headerEnabled = var.branding_header_enabled
    headerHtml    = var.branding_header_html
    footerEnabled = var.branding_footer_enabled
    footerHtml    = var.branding_footer_html
  }
}


**We have the following error**

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x104c9a954]

goroutine 81 [running]:
terraform-provider-sonatyperepo/internal/provider/model.(*UiBrandingCapabilityModel).FromApiModel(0x1400019ebd0?, 0x14000486188?)
 terraform-provider-sonatyperepo/internal/provider/model/capability_ui.go:49 +0x54
terraform-provider-sonatyperepo/internal/provider/capability/capability_type.(*UiBrandingCapability).UpdateStateFromApi(0x105449ca8?, {0x105322d40?, 0x140003e2a80?}, {0x1054011a0?, 0x140003ebb90?})
 terraform-provider-sonatyperepo/internal/provider/capability/capability_type/ui_branding.go:122 +0xbc
terraform-provider-sonatyperepo/internal/provider/capability.(*capabilityResource).Create(0x1400016abe0, {0x105449ca8, 0x1400038d5c0}, {{{{0x10544ef28, 0x140000ec330}, {0x1052c2900, 0x140000ec150}}, {0x105452220, 0x1400031e3c0}}, {{{0x10544ef28, ...}, ...}, ...}, ...}, ...)
 terraform-provider-sonatyperepo/internal/provider/capability/capability_common.go:108 +0x348
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).CreateResource(0x140002fc008, {0x105449ca8, 0x1400038d5c0}, 0x140002034f8, 0x140002034d0)
 github.com/hashicorp/terraform-plugin-framework@v1.14.1/internal/fwserver/server_createresource.go:101 +0x400
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ApplyResourceChange(0x140002fc008, {0x105449ca8, 0x1400038d5c0}, 0x1400016ac30, 0x140002035f0)
 github.com/hashicorp/terraform-plugin-framework@v1.14.1/internal/fwserver/server_applyresourcechange.go:57 +0x380
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ApplyResourceChange(0x140002fc008, {0x105449ca8?, 0x1400038cff0?}, 0x1400016ab40)
 github.com/hashicorp/terraform-plugin-framework@v1.14.1/internal/proto6server/server_applyresourcechange.go:55 +0x2e0
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0x14000224f00, {0x105449ca8?, 0x1400038c1e0?}, 0x140003f2070)
 github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov6/tf6server/server.go:866 +0x2a8
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x1054087c0, 0x14000224f00}, {0x105449ca8, 0x1400038c1e0}, 0x14000016000, 0x0)
 github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:611 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x1400021c200, {0x105449ca8, 0x1400038c150}, 0x140003e2060, 0x140002f28d0, 0x105c0cd98, 0x0)
 google.golang.org/grpc@v1.73.0/server.go:1405 +0xca8
google.golang.org/grpc.(*Server).handleStream(0x1400021c200, {0x10544a578, 0x14000135860}, 0x140003e2060)
 google.golang.org/grpc@v1.73.0/server.go:1815 +0x910
google.golang.org/grpc.(*Server).serveStreams.func2.1()
 google.golang.org/grpc@v1.73.0/server.go:1035 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 29
 google.golang.org/grpc@v1.73.0/server.go:1046 +0x13c

Error: The terraform-provider-sonatyperepo_v0.12.0 plugin crashed!

--- a/internal/provider/model/capability_audit.go
+++ b/internal/provider/model/capability_audit.go
@@ -31,9 +31,9 @@ type CapabilityAuditModel struct {
 }
 
 func (m *CapabilityAuditModel) FromApiModel(api *v3.CapabilityDTO) {
-	m.Id = types.StringValue(*api.Id)
-	m.Notes = types.StringValue(*api.Notes)
-	m.Enabled = types.BoolValue(*api.Enabled)
+	m.Id = types.StringPointerValue(api.Id)
+	m.Notes = types.StringPointerValue(api.Notes)
+	m.Enabled = types.BoolPointerValue(api.Enabled)
 }
 
 func (m *CapabilityAuditModel) ToApiCreateModel(version common.SystemVersion) *v3.CapabilityDTO {

--- a/internal/provider/model/capability_core.go
+++ b/internal/provider/model/capability_core.go
@@ -42,9 +42,9 @@ type CapabilityCoreBaseUrlModel struct {
 }
 
 func (m *CapabilityCoreBaseUrlModel) FromApiModel(api *v3.CapabilityDTO) {
-	m.Id = types.StringValue(*api.Id)
-	m.Notes = types.StringValue(*api.Notes)
-	m.Enabled = types.BoolValue(*api.Enabled)
+	m.Id = types.StringPointerValue(api.Id)
+	m.Notes = types.StringPointerValue(api.Notes)
+	m.Enabled = types.BoolPointerValue(api.Enabled)
 	m.Properties = &CapabilityPropertiesCoreBaseUrl{}
 	m.Properties.Url = types.StringValue((*api.Properties)["url"])
 }

--- a/internal/provider/model/capability_firewall.go
+++ b/internal/provider/model/capability_firewall.go
@@ -43,9 +43,9 @@ type CapabilityFirewallAuditQuarantineModel struct {
 }
 
 func (m *CapabilityFirewallAuditQuarantineModel) FromApiModel(api *v3.CapabilityDTO) {
-	m.Id = types.StringValue(*api.Id)
-	m.Notes = types.StringValue(*api.Notes)
-	m.Enabled = types.BoolValue(*api.Enabled)
+	m.Id = types.StringPointerValue(api.Id)
+	m.Notes = types.StringPointerValue(api.Notes)
+	m.Enabled = types.BoolPointerValue(api.Enabled)
 	m.Properties = &CapabilityPropertiesFirewallAuditQuarantine{}
 	m.Properties.Quarantine = types.BoolValue(ParseBool(
 		(*api.Properties)["quarantine"],

--- a/internal/provider/model/capability_healthcheck.go
+++ b/internal/provider/model/capability_healthcheck.go
@@ -43,9 +43,9 @@ type CapabilityHealthcheckModel struct {
 }
 
 func (m *CapabilityHealthcheckModel) FromApiModel(api *v3.CapabilityDTO) {
-	m.Id = types.StringValue(*api.Id)
-	m.Notes = types.StringValue(*api.Notes)
-	m.Enabled = types.BoolValue(*api.Enabled)
+	m.Id = types.StringPointerValue(api.Id)
+	m.Notes = types.StringPointerValue(api.Notes)
+	m.Enabled = types.BoolPointerValue(api.Enabled)
 	m.Properties = &CapabilityPropertiesHealthcheck{}
 	m.Properties.ConfiguredForAllProxies = types.BoolValue(ParseBool(
 		(*api.Properties)["configuredForAll"],

--- a/internal/provider/model/capability_s3.go
+++ b/internal/provider/model/capability_s3.go
@@ -44,9 +44,9 @@ type CapabilityCustomS3RegionsModel struct {
 }
 
 func (m *CapabilityCustomS3RegionsModel) FromApiModel(api *v3.CapabilityDTO) {
-	m.Id = types.StringValue(*api.Id)
-	m.Notes = types.StringValue(*api.Notes)
-	m.Enabled = types.BoolValue(*api.Enabled)
+	m.Id = types.StringPointerValue(api.Id)
+	m.Notes = types.StringPointerValue(api.Notes)
+	m.Enabled = types.BoolPointerValue(api.Enabled)
 	m.Properties = &CapabilityPropertiesCustomS3Regions{}
 
 	// Regions

--- a/internal/provider/model/capability_security.go
+++ b/internal/provider/model/capability_security.go
@@ -42,9 +42,9 @@ type CapabilityRutAuthModel struct {
 }
 
 func (m *CapabilityRutAuthModel) FromApiModel(api *v3.CapabilityDTO) {
-	m.Id = types.StringValue(*api.Id)
-	m.Notes = types.StringValue(*api.Notes)
-	m.Enabled = types.BoolValue(*api.Enabled)
+	m.Id = types.StringPointerValue(api.Id)
+	m.Notes = types.StringPointerValue(api.Notes)
+	m.Enabled = types.BoolPointerValue(api.Enabled)
 	m.Properties = &CapabilityPropertiesRutAuth{}
 	m.Properties.HttpHeader = types.StringValue((*api.Properties)["http_header"])
 }

--- a/internal/provider/model/capability_ui.go
+++ b/internal/provider/model/capability_ui.go
@@ -45,9 +45,9 @@ type UiBrandingCapabilityModel struct {
 }
 
 func (m *UiBrandingCapabilityModel) FromApiModel(api *v3.CapabilityDTO) {
-	m.Id = types.StringValue(*api.Id)
-	m.Notes = types.StringValue(*api.Notes)
-	m.Enabled = types.BoolValue(*api.Enabled)
+	m.Id = types.StringPointerValue(api.Id)
+	m.Notes = types.StringPointerValue(api.Notes)
+	m.Enabled = types.BoolPointerValue(api.Enabled)
 	m.Properties = &CapabilityPropertiesUiBranding{}
 	m.Properties.FooterEnabled = types.BoolValue(ParseBool(
 		(*api.Properties)["footerEnabled"],

--- a/internal/provider/model/capability_webhook.go
+++ b/internal/provider/model/capability_webhook.go
@@ -70,9 +70,9 @@ type WebhookGlobalCapabilityModel struct {
 }
 
 func (m *WebhookGlobalCapabilityModel) FromApiModel(api *v3.CapabilityDTO) {
-	m.Id = types.StringValue(*api.Id)
-	m.Notes = types.StringValue(*api.Notes)
-	m.Enabled = types.BoolValue(*api.Enabled)
+	m.Id = types.StringPointerValue(api.Id)
+	m.Notes = types.StringPointerValue(api.Notes)
+	m.Enabled = types.BoolPointerValue(api.Enabled)
 	m.Properties = &CapabilityPropertiesWebhookGlobal{}
 	namesStr := (*api.Properties)["names"]
 	namesParts := strings.Split(namesStr, ",")

